wordpress-mobile/WordPress-iOS
# Comments: Add author ID field.

Refs #17476 
Depends on https://github.com/wordpress-mobile/WordPressKit-iOS/pull/469 👈 Please review this first, thank you!

This PR adds an `authorID` field, which represents the comment author's dotcom user ID. For context, this field will be used later on to show an "Author" badge in the comment thread if the comment is written by the post author (see #17476 for more details). Some methods in `CommentService` are slightly adjusted to ensure that the `authorID` is properly parsed to/from `RemoteComment`.

NOTE: I'll update the Podfile to point to WordPressKit `4.44.0-beta.2` once https://github.com/wordpress-mobile/WordPressKit-iOS/pull/469 is merged.

## To test

Since the field is not used yet, we'll need to ensure that this change does not introduce any side effects. Here are a couple of test cases that I've used:

- Go to My Site > Comments, and scroll through the list. Verify that the comments are displayed correctly.
- Tap one of the comments. Verify that the comment detail lists the comment correctly.
- Submit a reply to the comment. Verify that a reply is successfully submitted. 
- Go to Notifications, and select the comments filter. Verify that the comments are displayed correctly.
- Go to Reader, and tap on the comment icon. Verify that the thread displays the comments correctly.
- Go back to Reader, and this time try to submit a comment on the post. Verify that the comment is submitted successfully.

## Regression Notes
1. Potential unintended areas of impact
Should be none. The new field is not used anywhere yet.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually smoke-tested  the comments flow.

3. What automated tests I added (or what prevented me from doing so)
Modified `CommentServiceRemoteREST` tests in WordPressKit to account for `authorID`.

PR submission checklist:
